### PR TITLE
Issue #DP-30 feat: Partial textbook

### PR DESCRIPTION
**Pull request by ahghatol**

### CODE REVIEW

File: src/service/programService.js [hunk 1]
```diff
@@ -1394,13 +1394,21 @@ async function programCopyCollections(req, response) {
   }
 
   const collections = _.get(data, 'request.collections');
+  // const primaryCollectionId = _.get(data, 'request.collections')[];
+  const collectionIds = [];
+
+  _.forEach(collections, (item) => {
+    collectionIds.push(item.id);
+  });
+
```
Comment (venkateshwarans) [inline src/service/programService.js:1403]: can we rewrite it to const collectionIds = _.map(collections, 'id') ?

File: src/service/programService.js [hunk 2]
```diff
@@ -1419,7 +1429,9 @@ async function programCopyCollections(req, response) {
                   return _.get(r, 'data')
                 })
                 const getCollectiveRequest = _.map(originHierarchyResultData, c => {
-                  return hierarchyService.existingHierarchyUpdateRequest(c, additionalMetaData);
+                  const cindex = collections.findIndex(r => r.id === c.hierarchy.content.identifier);
+                  const children = collections[cindex].children;
```
Comment (venkateshwarans) [inline src/service/programService.js:1433]: can we have a check if `index !== -1`

File: src/service/programService.js [hunk 3]
```diff
@@ -1473,8 +1486,11 @@ async function programCopyCollections(req, response) {
                       return mapOriginalHierarchy;
                     })
                     const getBulkUpdateRequest = _.map(createdCollections, item => {
-                      return hierarchyService.newHierarchyUpdateRequest(item, additionalMetaData)
+                      const cindex = collections.findIndex(r => r.id === item.hierarchy.content.identifier);
+                      const children = collections[cindex].children;
```
Comment (venkateshwarans) [inline src/service/programService.js:1490]: same here

File: src/service/programService.js (hunk 1 above)
Comment (ahghatol) [reply src/service/programService.js]: Done

File: src/service/programService.js (hunk 2 above)
Comment (ahghatol) [reply src/service/programService.js]: Done

File: src/service/programService.js (hunk 3 above)
Comment (ahghatol) [reply src/service/programService.js]: Done

**Comment by ahghatol** [comment]
Done

**Pull request merged**